# Домашнее задание № 1

## по дисциплине: Теория вероятностей и математическая статистика

## Тема: Случайные события. Условная вероятность. Формула Байеса.  Независимые испытания

### выполнил: Евгений Кириллов

In [1]:
import numpy as np

__Задача 1__

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

воспользуемся формулой размещений:

### $$A_n^k = \frac{n!}{(n - k)!}$$

In [2]:
def arrangements(n, k):
    return np.math.factorial(n) // np.math.factorial(n - k)

### $$A_{10}^4 = \frac{10!}{(10 - 4)!}=7\cdot 8\cdot 9\cdot 10=5040$$

In [3]:
arrangements(10, 4)

5040

__Задача 2__

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

воспользуемся формулой сочетаний:

### $$C_n^k = \frac{n!}{k!(n - k)!}$$

In [4]:
def combinations(n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

чтобы решить эту задачу, нужно сначала рассмотреть ситуацию, когда выбирается один туз из четырёх:

### $$C_{4}^1 = \frac{4!}{1!(4 - 1)!}=4$$

In [5]:
combinations(4, 1)

4

остальные три карты выбираются из 48 карт — тузы не рассматриваются, так как они уже выбраны:

### $$C_{48}^3 = \frac{48!}{3!(48 - 3)!}=\frac{48!}{3!(45)!}=\frac{46\cdot 47\cdot 48}{3!}=\frac{46\cdot 47\cdot 48}{6}=17296$$

In [6]:
combinations(48, 3)

17296

In [7]:
a = combinations(4, 1) * combinations(48, 3)

print(a)

69184


аналогично считаются ситуации, когда тузов 2, 3 и 4:

In [8]:
b = combinations(4, 2) * combinations(48, 2)
c = combinations(4, 3) * combinations(48, 1)
d = combinations(4, 4) * combinations(48, 0)

print(b)
print(c)
print(d)

6768
192
1


итоговым ответом на задание будет сумма всех рассчитанных выше сочетаний:

In [9]:
a + b + c + d

76145

__Задача 3__

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

упростим задачу, вычислим для начала вероятность того, что два определённых человека будут сидеть рядом слева, т.е. на 1-й и 2-й позиции скамейки:

в таком случае получается, что количество всех возможных вариантов рассадки 7 человек на скамейке будет вычисляться по формуле перестановок:

### $$P_n = n!$$

In [10]:
def permutations(n):
    return np.math.factorial(n)

### $$P_7 = 7!=5040$$

In [11]:
permutations(7)

5040

а количество вариантов при которых два определённых человека будут сидеть на 1-й и 2-й позиции скамейки, а остальные справа от них, будет вычислено по формуле:

### $$P_2\cdot P_5 = 2!\cdot 5!=240$$

In [12]:
permutations(2) * permutations(5)

240

соответственно, вероятность того, что два определённых человека будут сидеть рядом слева, т.е. на 1-й и 2-й позиции скамейки вычисляется по формуле:

### $$P = \frac{m}{n}=\frac{240}{5040}\approx0.0476190476190476$$

т.к. на скамейке для семи человек имеется 6 вариантов размещения двух людей для того, чтобы они сидели вместе (без учёта порядка в котором сидят эти два человека), то тогда итоговая формула для решения задачи приобретает следующий вид:

### $$P=\frac{6\cdot P_2\cdot P_5}{P_7}=\frac{6\cdot 240}{5040}\approx 0.2857142857142857$$

проверим ответ численно:

In [13]:
# считаем все возможные исходы
all_outcomes = [str(i) for i in range(1234567, 7654322)]
for i in range(1, 8):
    all_outcomes = [ii for ii in all_outcomes if ii.count(str(i)) == 1]

# считаем исходы которые удовлетворяют условию задания
required_outcomes = []
for i in all_outcomes:
    try:
        if i[list(i).index('1') - 1] == '2' and list(i).index('1') - 1 >= 0:
            required_outcomes.append(i)
            continue
    except:
        None

    try:
        if i[list(i).index('1') + 1] == '2':
            required_outcomes.append(i)
            continue
    except:
        None

# считаем вероятность
probability = len(required_outcomes) / len(all_outcomes)
probability

0.2857142857142857

всё верно!

__Задача 4__

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

посчитаем количество всех возможных вариантов, для чего используем формулу сочетаний:

### $$C_n^k = \frac{n!}{k!(n - k)!}$$

In [14]:
def combinations(n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

### $$C_{60}^3 = \frac{60!}{3!(60 - 3)!}=\frac{58\cdot 59\cdot 60}{6}=34220$$

In [15]:
combinations(60, 3)

34220

теперь посчитаем количество сочетаний билетов, которые знает студент:

### $$C_{50}^3 = \frac{50!}{3!(50 - 3)!}=\frac{48\cdot 49\cdot 50}{6}=19600$$

In [16]:
combinations(50, 3)

19600

посчитаем вероятность того, что студент вытянет все билеты, которые он знает:

### $$P = \frac{m}{n}=\frac{C_{50}^3}{C_{60}^3}=\frac{19600}{34220}\approx0.5727644652250146$$

теперь посчитаем количество сочетаний билетов, когда студент знает только 2 билета:

### $$C_{50}^2\cdot C_{10}^1 = \frac{50!}{2!(50 - 2)!}\cdot 10=\frac{49\cdot 50}{2}\cdot 10=12250$$

In [17]:
combinations(50, 2) * combinations(10, 1)

12250

посчитаем вероятность того, что студент вытянет 2 билета, которые он знает:

### $$P = \frac{m}{n}=\frac{C_{50}^2\cdot C_{10}^1}{C_{60}^3}=\frac{12250}{34220}\approx0.3579777907656341$$

In [18]:
combinations(50, 2) * combinations(10, 1) / combinations(60, 3)

0.35797779076563413

ответ:

a) 0.5727644652250146

б) 0.3579777907656341

__Задача 5__

Бросается игральная кость. Пусть событие `A` - появление чётного числа, событие `B` - появление числа больше трёх. Являются ли эти события независимыми?

ответ: нет, эти события зависимые, т.к. кость бросается только один раз

__Задача 6__

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1 % населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99 % больных этой болезнью и всего лишь в 1 % случаев даёт ложный положительный ответ.

_Вопрос_: какова вероятность, что вы действительно больны ей?

_Подсказка_: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

ответ: Т.к. тест является положительным, то уже нет никакой надобности принимать во внимание факт, что очень редкая болезнь поражает 0.1 % населения. Соответственно, вероятность того, что человек с положительным тестом болен = 0.99